diff --git a/configure.ac b/configure.ac
index ecf8384bf..545d0cd03 100644
--- a/configure.ac
+++ b/configure.ac
@@ -632,6 +632,8 @@ AS_IF([test "x$enable_capstone" != xno], [
 		fi
 		enable_capstone=no
 	])
+], [
+	AC_DEFINE([HAVE_CAPSTONE], [0], [0 if you don't have Capstone disassembly framework.])
 ])
 
 AS_IF([test "x$enable_capstone" == xno], [
diff --git a/src/flash/nor/Makefile.am b/src/flash/nor/Makefile.am
index 534a7a804..159244502 100644
--- a/src/flash/nor/Makefile.am
+++ b/src/flash/nor/Makefile.am
@@ -27,6 +27,7 @@ NOR_DRIVERS = \
 	%D%/cfi.c \
 	%D%/dsp5680xx_flash.c \
 	%D%/efm32.c \
+	%D%/efm32s2.c \
 	%D%/em357.c \
 	%D%/esirisc_flash.c \
 	%D%/faux.c \
diff --git a/src/flash/nor/driver.h b/src/flash/nor/driver.h
index a63b72c8f..b2af2c6f0 100644
--- a/src/flash/nor/driver.h
+++ b/src/flash/nor/driver.h
@@ -255,6 +255,7 @@ extern const struct flash_driver cc3220sf_flash;
 extern const struct flash_driver cfi_flash;
 extern const struct flash_driver dsp5680xx_flash;
 extern const struct flash_driver efm32_flash;
+extern const struct flash_driver efm32s2_flash;
 extern const struct flash_driver em357_flash;
 extern const struct flash_driver esirisc_flash;
 extern const struct flash_driver faux_flash;
diff --git a/src/flash/nor/drivers.c b/src/flash/nor/drivers.c
index 3157bd329..11fa01c4e 100644
--- a/src/flash/nor/drivers.c
+++ b/src/flash/nor/drivers.c
@@ -32,6 +32,7 @@ static const struct flash_driver * const flash_drivers[] = {
 	&cfi_flash,
 	&dsp5680xx_flash,
 	&efm32_flash,
+	&efm32s2_flash,
 	&em357_flash,
 	&esirisc_flash,
 	&faux_flash,
