diff --git a/configure.ac b/configure.ac
index bd2e5c4f9..3034c10b3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -600,6 +600,8 @@ AS_IF([test "x$enable_capstone" != xno], [
 		fi
 		enable_capstone=no
 	])
+], [
+	AC_DEFINE([HAVE_CAPSTONE], [0], [0 if you don't have Capstone disassembly framework.])
 ])
 
 for hidapi_lib in hidapi hidapi-hidraw hidapi-libusb; do
diff --git a/src/flash/nor/Makefile.am b/src/flash/nor/Makefile.am
index 532670436..a0b4a8390 100644
--- a/src/flash/nor/Makefile.am
+++ b/src/flash/nor/Makefile.am
@@ -25,6 +25,7 @@ NOR_DRIVERS = \
 	%D%/cfi.c \
 	%D%/dsp5680xx_flash.c \
 	%D%/efm32.c \
+	%D%/efm32s2.c \
 	%D%/em357.c \
 	%D%/esirisc_flash.c \
 	%D%/faux.c \
diff --git a/src/flash/nor/drivers.c b/src/flash/nor/drivers.c
index 6eadc756b..d0bc60573 100644
--- a/src/flash/nor/drivers.c
+++ b/src/flash/nor/drivers.c
@@ -38,6 +38,7 @@ extern const struct flash_driver cc26xx_flash;
 extern const struct flash_driver cfi_flash;
 extern const struct flash_driver dsp5680xx_flash;
 extern const struct flash_driver efm32_flash;
+extern const struct flash_driver efm32s2_flash;
 extern const struct flash_driver em357_flash;
 extern const struct flash_driver esirisc_flash;
 extern const struct flash_driver faux_flash;
@@ -112,6 +113,7 @@ static const struct flash_driver * const flash_drivers[] = {
 	&cfi_flash,
 	&dsp5680xx_flash,
 	&efm32_flash,
+	&efm32s2_flash,
 	&em357_flash,
 	&esirisc_flash,
 	&faux_flash,
